### ISSUE TITLE
Making a SlicePlot from a cut_region doesn't work

### ISSUE THREAD

**Issue by ngoldbaum** [yt-project/yt#1646]
### Bug report

**Bug summary**

Creating a SlicePlot using a cut_region on randomly generated data and `load_uniform_grid` produces a blank plot, despite the cut_region selecting all data.

**Code for reproduction**

```python
import yt
import numpy as np

np.random.seed(0x4d3d3d3)

arr = np.random.random(size=(41, 56, 47))
data = dict(density=(arr, "g/cm**3"))
bbox = np.array([[-4.0, 4.2], [-5.5, 5.7], [-4.175872, 5.22413]])
ds = yt.load_uniform_grid(data, arr.shape, length_unit="cm", bbox=bbox)

ad = ds.all_data()
print(ad.quantities.extrema([("gas", "density")]))
p = yt.SlicePlot(ds, 'x', 'density', center="m", data_source=ad)
p.save('ad.png')

den = ad.cut_region(['obj["density"] > 0.0'])
print(den.quantities.extrema([("gas", "density")]))

p = yt.SlicePlot(ds, 'x', 'density', center="m", data_source=den)
p.save('cut.png')
```

**Actual outcome**

The image cut.png should match ad.png.

**Expected outcome**

ad.png: https://imgur.com/a/4wIxB

cut.png: https://imgur.com/a/D608d

**Version Information**

  * Operating System: MacOS
  * Python Version: 3.6
  * yt version: 3.5dev
  * Other Libraries (if applicable):

**Comment by ngoldbaum** [comment]
I thought that I had fixed this when we closed #941. Unfortunately I think this is getting triggered by the exact floating point comparisons that I had originally thought was causing trouble.

We should probably go back to the solution proposed in #1352 (use relatively fine morton indices instead of exact floating comparison) which should allow us a bit of floating point slop.

**Comment by matthewturk** [comment]
I just ran this script, and it provides identical results for `ad.png` and `cut.png` for me.  So I am going to close this.
